test/functional/p2p_sendheaders.py

```diff
@@ -174,7 +171,7 @@ def check_last_headers_announcement(self, headers):
         """Test whether the last headers announcements received are right.
            Headers may be announced across more than one message."""
         test_function = lambda: (len(self.recent_headers_announced) >= len(headers))
-        wait_until(test_function, timeout=60, lock=mininode_lock)
+        self.wait_until(test_function)
```

Comment:
why are you removing the timeout and lock here?

Reply:
This class extends `P2PInterface` not `BitcoinTestFramework`. mininode one already has a lock. As for timeout, it already has a default value of 60.